python/tvm/script/parser/tir/entry.py

```diff
@@ -50,6 +50,48 @@ def prim_func(func: Callable) -> Union[PrimFunc, Callable]:
 setattr(prim_func, "dispatch_token", "tir")
 
 
+# Semantics of TIR macros:
+# - Function that is decorated with @T.macro can have any parameters that
+#   follow Python syntax, i.e. positional, keyword, etc. Type annotations
+#   are not required, but are allowed.
+# - The arguments to `T.insert` are: macro name (either as value, or as
+#   a string with the name), followed by the argument list.
+#   For `T.insert(arg1, arg2, arg3, ...)`, the values are substituted into
+#   the body of the macro as in the call `arg1(arg2, arg3, ...)`.
+#   The body with the substituted values is then inserted at the point
+#   where the `T.insert` is located.
+
+
+class TIRMacro:
```

Comment:
Would it be `I.macro` then?

Reply:
It would, yes.  Within TVMScript, the default set of global definitions is defined [here](https://github.com/apache/tvm/blob/main/python/tvm/script/parser/core/entry.py#L44).  This provides both `tvm.script.tir` as `T` and `tvm.script.ir` as `I`.

Reply:
Can we do this in a separate PR?